crates/gettext-maps/build.rs

```diff
@@ -0,0 +1,138 @@
+use std::{
+    env,
+    ffi::OsString,
+    path::{Path, PathBuf},
+    process::Command,
+};
+
+fn main() {
+    let cargo_target_dir = fish_build_helper::get_target_dir();
+    // FISH_BUILD_DIR is set by CMake, if we are using it.
+    let fish_build_dir =
+        option_env!("FISH_BUILD_DIR").unwrap_or(cargo_target_dir.to_str().unwrap());
+    let cache_dir = PathBuf::from(fish_build_dir).join("fish-localization-map-cache");
+    embed_localizations(&cache_dir);
+
+    fish_build_helper::rebuild_if_path_changed(fish_build_helper::get_repo_root().join("po"));
+}
+
+fn embed_localizations(cache_dir: &Path) {
+    use fish_mo_file_parser::parse_mo_file;
+    use std::{
+        fs::File,
+        io::{BufWriter, Write},
+    };
+
+    let po_dir = fish_build_helper::get_repo_root().join("po");
+
+    // Ensure that the directory is created, because clippy cannot compile the code if the
+    // directory does not exist.
+    std::fs::create_dir_all(cache_dir).unwrap();
+
+    let localization_map_path =
+        Path::new(&env::var("OUT_DIR").unwrap()).join("localization_maps.rs");
+    let mut localization_map_file = BufWriter::new(File::create(&localization_map_path).unwrap());
+
+    // This will become a map which maps from language identifiers to maps containing localizations
+    // for the respective language.
+    let mut catalogs = phf_codegen::Map::new();
+
+    match Command::new("msgfmt").arg("-h").status() {
+        Err(e) if e.kind() == std::io::ErrorKind::NotFound => {
+            println!("cargo:warning=Cannot find msgfmt to build gettext message catalogs. Localization will not work.");
```

Comment:
FWIW we have `rsconf::warn!` for this.
It seems like a readable abstraction (though hard to discover
for people who already know the cargo:warning trick), so I guess I'd
update this and crates/build-man-pages/build.rs.
Not a strong opinion though